### ISSUE TITLE
Rule suggestion: Make sure non-optional class members are initialized 

### ISSUE THREAD

**Issue by mohsen1** [palantir/tslint#2063]
```ts
class Foo {
  bar: string;
  constructor() {
  }
}
```

`bar` is not initialized in constructor but `bar` is not type `string | undefined`. Linter should warn about this.

**Comment by mohsen1** [comment]
Dupe: https://github.com/palantir/tslint/issues/1414
